### ISSUE TITLE
DAG does not work well.

### ISSUE THREAD

**Issue by jstudy** [vlfeat/matconvnet#496]
I met a problem related to DAG. I found that the same CNN structure implementd using the DAG model alwasy had a higher test error than that uing the simple model. More confusingly, their training errors are almost the same.  I tried the NIN struture on CIFAR 10 dataset.  How could I solve the problem?

**Comment by bazilas** [comment]
Do you say that you got different results, between simpleNN and dagNN, by running the sample example (e.g. CIFAR 10) with one of the given architectures (e.g. NIN)?

**Comment by jstudy** [comment]
Yes, I used the NIN structure. But I revised it by adding  BN after each convolutioanl layer (see attachment ). SimpleNN  outperformed DagNN after running 169 epoches, with learning rate  as [0.5*ones(1,50)  0.5:-0.005:0.005  0.005:-0.00025:0.0005].
[cnn_cifar_init_nin_1.txt](https://github.com/vlfeat/matconvnet/files/219087/cnn_cifar_init_nin_1.txt)

**Comment by jstudy** [comment]
Hi, bazilas:
     You have released the pretrained 'googlenet-dag.mat'. But it is still not convinent to write the 'googlenet-dag.m' file directly by using the fromSimpleNN function. Would you like to update the codes?
       Sincerely,
         Shawn.

**Comment by lenck** [comment]
Hi, there was a bug in the bnorm dag wrapper, can you try it with the `devel` branch? It should be fixed there...
And I don't understand your question about the `googlenet-dag.m`. I guess, you mean if we have a source code which would define a googlenet network? The `.mat` files provided, unless it is a `matconvnet` network, are imported from caffe, so there is no definition file for them... But we are working on some new models now, so that may change in the near future.

**Comment by jstudy** [comment]
@lenck 
Thank you. 
Yes, you got it .  You have released the  'cnn_imagenet_init.m' file which  directly defines several famous networks. But there is on such a file which directly defines a googlenet network. I really expect your  new model for googlenet.
  Sincerely,
Shawn.
